Branch-SDK/src/main/java/io/branch/referral/ServerRequestQueue.java

```diff
@@ -71,29 +71,24 @@ private ServerRequestQueue(Context c) {
     }
     
     private void persist() {
-        new Thread(new Runnable() {
-            @Override
-            public void run() {
-                JSONArray jsonArr = new JSONArray();
-                synchronized (reqQueueLockObject) {
-                    for (ServerRequest aQueue : queue) {
-                        if (aQueue.isPersistable()) {
-                            JSONObject json = aQueue.toJSON();
-                            if (json != null) {
-                                jsonArr.put(json);
-                            }
+        try {
+            JSONArray jsonArr = new JSONArray();
+            synchronized (reqQueueLockObject) {
+                for (ServerRequest aQueue : queue) {
+                    if (aQueue.isPersistable()) {
+                        JSONObject json = aQueue.toJSON();
+                        if (json != null) {
+                            jsonArr.put(json);
                         }
                     }
                 }
-
-                try {
-                    editor.putString(PREF_KEY, jsonArr.toString()).commit();
```

Comment:
This changes from writing directly to disk to changing the shared prefs singleton + setting up an async operation to write the new value to disk. Previous implementation was fine because it was running in a separate thread, but to improve performance, I removed the creation of the thread. Not only will this improve performance, I think might also prevent concurrency bugs because thread creation takes time. To avoid any unexpected bugs that expanded the scope of the existing try catch block to include the entire method implementation.